src/coreclr/pal/inc/pal_mstypes.h

```diff
@@ -574,8 +574,8 @@ typedef long ptrdiff_t;
 typedef unsigned int size_t;
 typedef int ptrdiff_t;
 #endif // !HOST_64BIT
-#endif // !PAL_STDCPP_COMPAT
 #define _SIZE_T_DEFINED
+#endif // !PAL_STDCPP_COMPAT
```

Comment:
What's the impact range of this?

Reply:
What do you mean by "impact range"?

The line should affect all platforms. However, I believe that this is how things should be, since with `PAL_STDCPP_COMPAT` the header should not define things that potentially mess up system headers.

Some more comments about this change here: https://github.com/dotnet/runtime/issues/55803#issuecomment-1537114390